File: tests/codegen/snapshots/python/interface_fragments.py
from typing import Union

class OperationNameResultInterfaceBlogPost:
    # typename: BlogPost
    id: str
    title: str

class OperationNameResultInterfaceImage:
    # typename: Image
    id: str
    url: str

OperationNameResultInterface = Union[OperationNameResultInterfaceBlogPost, OperationNameResultInterfaceImage]

class OperationNameResult:
    interface: OperationNameResultInterface

File: tests/codegen/snapshots/typescript/interface_fragments.ts
type OperationNameResultInterfaceBlogPost = {
    id: string
    title: string
}

type OperationNameResultInterfaceImage = {
    id: string
    url: string
}

type OperationNameResultInterface = OperationNameResultInterfaceBlogPost | OperationNameResultInterfaceImage

type OperationNameResult = {
    interface: OperationNameResultInterface
}
